src/ReferenceList.php

```diff
@@ -217,10 +217,10 @@ public function serialize() {
 	 *
 	 * @since 2.1
 	 *
-	 * @param string $data
+	 * @param string $serialized
 	 */
-	public function unserialize( $data ) {
-		$this->__construct( unserialize( $data ) );
+	public function unserialize( $serialized ) {
+		$this->references = unserialize( $serialized );
```

Comment:
Yes, that's the intention of this patch. `unserialize` must not do validation that was already done when the original object was created. Note that PHP only calls this method with serialization strings that are clearly marked as `ReferenceList` serializations.